### DISCUSSION TITLE
My Kong service routes do not seem to apply any regex expression whatsoever...

### DISCUSSION THREAD

**Discussion by PRBITNA** [Kong/kong#9545] [Help] [answered]
HI,

I am using Kong 3.0.0-alpine (via Docker image). I can get services and routes configured easily and they work fine, but when I try and put regex expressions to add wildcards to my route paths - I simply cannot get it going. I have read the official documentation and many forums.... I have tried various setups all with the same results:

-Kong/with postgreSQL DB: added service and routes via curl commands
-Kong/DBless: added service and route via declarative kong.yaml 
-Kong/with postgreSQl DB/Konga: added service and route via Konga UI

I am probably doing something wrong/stupid (please tell me), but I cant spot it. Sample config below (various regex expressions tried):

without DB and kong.yaml:

 _format_version: "3.0"
 _transform: true

 services:
 - host: mockbin.org
   name: example_service
   port: 80
   protocol: http
   routes:
   - name: example_route
     paths: ['/items/.+']   <=== have tried various methods of declaring here
     strip_path: true

with DB and curl: (also tried many different methods of declaring paths)
curl -i -X POST http://localhost:8001/services/example_service/routes --data name=example_route_regex --data-urlencode paths[]=/status/\d+
curl -i -X POST http://localhost:8001/services/example_service/routes --data name=example_route_regex --data-urlencode paths=/paul/.+

what ends up happening is that the regex just gets taken literally in the route URL - i.e when trying to use '.+' regex in the route definition...  the route actually works and the service/route  is available at   http://localhost:8000/items/.+  !!

Any corrections or opinions welcome.

**Comment by chronolaw** [comment, marked as answer]
In Kong 3.0, we must add a `~` prefix to declare a regex path, it is a breaking change.

I think that you could try `~/items/.+`

Please refer to our change log. https://github.com/Kong/kong/blob/master/CHANGELOG.md#300

**Reply by PRBITNA** [reply]
Legend thanks.

I read everything except those release notes!

"Kong will no longer use an heuristic to guess whether a route.path is a regex pattern. From now 3.0 onwards, all regex paths must start with the "~" prefix, and all paths that don't start with "~" will be considered plain text. The migration process should automatically convert the regex paths when upgrading from 2.x to 3.0 https://github.com/Kong/kong/pull/9027"
